### PR TITLE
SREP-1792: Error if sts token output is nil

### PR DESCRIPTION
**Pull request by fahlmant**
I have been unable to replicate this part segfaulting. Adding in some protection to make sure that a nil point is not accessed.

### CODE REVIEW

**Comment by jewzaam** [comment]
Is this an intermittent issue or something that's failing in a specific scenario?  It sounds like this will keep the operator from crashing.  But it may mask issues with processing the STS credentials.  Do we have alerts on these tokens being stale or not updated?

**Comment by fahlmant** [comment]
@jewzaam I haven't been able to recreate the issue, seem intermittent on stage. 

We have a credentials rotator to make sure these are fresh. I'm not sure an alert is the right approach for that, but that's out of scope of what this PR is trying to accomplish.

**Comment by rafael-azevedo** [comment]
> Is this an intermittent issue or something that's failing in a specific scenario? It sounds like this will keep the operator from crashing. But it may mask issues with processing the STS credentials. Do we have alerts on these tokens being stale or not updated?

The issue processing the STS token is most likely api flakiness. By breaking here in the future refactor we should be able to start from this step again should it break here instead of fail the account.

File: pkg/controller/account/account_controller.go [hunk 1]
```diff
@@ -129,6 +129,9 @@ var ErrAwsSupportCaseIDNotFound = errors.New("SupportCaseIdNotfound")
 // ErrAwsFailedDescribeSupportCase indicates that the support case describe failed
 var ErrAwsFailedDescribeSupportCase = errors.New("FailedDescribeSupportCase")
 
+//ErrFederationTokenOutputNil indicates that getting a federation token from AWS failed
```
Comment (jharrington22) [inline pkg/controller/account/account_controller.go]: Nit pick but could you add a space before the text?

**Comment by jharrington22** [comment]
> > Is this an intermittent issue or something that's failing in a specific scenario? It sounds like this will keep the operator from crashing. But it may mask issues with processing the STS credentials. Do we have alerts on these tokens being stale or not updated?
> 
> The issue processing the STS token is most likely api flakiness. By breaking here in the future refactor we should be able to start from this step again should it break here instead of fail the account.

Since we are returning an error now the operator will retry this step since its before we set `status.rotateCredentials` to `false`

**Pull request merged**